test/integration/client-side-encryption/client_side_encryption.prose.25.lookup.test.ts

```diff
@@ -0,0 +1,413 @@
+          options: { validator: { $jsonSchema: await readFixture('schema-csfle2.json') } },
+          document: { csfle2: 'csfle2' }
+        },
+        {
+          name: 'qe',
+          options: { encryptedFields: await readFixture('schema-qe.json') },
+          document: { qe: 'qe' }
+        },
+        {
+          name: 'qe2',
+          options: { encryptedFields: await readFixture('schema-qe2.json') },
+          document: { qe2: 'qe2' }
+        },
+        {
+          name: 'no_schema',
+          options: {},
+          document: { no_schema: 'no_schema' }
+        },
+        {
+          name: 'no_schema2',
+          options: {},
+          document: { no_schema2: 'no_schema2' }
+        }
+      ];
+
+      for (const { name, options } of collections) {
+        await client.db('db').createCollection(name, options);
+      }
+
+      /**
+       * Create an encrypted MongoClient configured with:
+       *
+       * ```txt
+       *   AutoEncryptionOpts(
+       *       keyVaultNamespace="db.keyvault",
+       *       kmsProviders={"local": { "key": "<base64 decoding of LOCAL_MASTERKEY>" }}
+       *   )
+       * ```
+       */
+      encryptedClient = newEncryptedClient(this);
+
+      /**
+       * ```
+       * {"csfle": "csfle"} into db.csfle
+       * Use the unencrypted client to retrieve it. Assert the csfle field is BSON binary.
+       * {"csfle2": "csfle2"} into db.csfle2
+       * Use the unencrypted client to retrieve it. Assert the csfle2 field is BSON binary.
+       * {"qe": "qe"} into db.qe
+       * Use the unencrypted client to retrieve it. Assert the qe field is BSON binary.
+       * {"qe2": "qe2"} into db.qe2
+       * Use the unencrypted client to retrieve it. Assert the qe2 field is BSON binary.
+       * {"no_schema": "no_schema"} into db.no_schema
+       * {"no_schema2": "no_schema2"} into db.no_schema2
+       * ```
+       */
+      for (const { name, document } of collections) {
+        const { insertedId } = await encryptedClient.db('db').collection(name).insertOne(document);
+
+        if (name.startsWith('no_')) continue;
+
+        expect(await client.db('db').collection(name).findOne(insertedId))
+          .to.have.property(Object.keys(document)[0])
+          .that.has.property('_bsontype', 'Binary');
+      }
+    } finally {
+      await client?.close();
+      await encryptedClient?.close();
+    }
+  });
+
+  const test = function (
+    title: string,
+    collName: string,
+    pipeline: Document[],
+    expected: Document,
+    metadata?: MongoDBMetadataUI
+  ) {
+    describe(title.slice(0, title.indexOf(':')), function () {
+      let client: MongoClient;
+
+      beforeEach(async function () {
+        Error.stackTraceLimit = 0;
+        client = newEncryptedClient(this);
+      });
+
+      afterEach(async function () {
+        await client.close();
+      });
+
+      it(title.slice(title.indexOf(':') + 1).trim(), metadata ?? defaultMetadata, async () => {
```

Comment:
Similar to the above, how would you feel about just writing 9 test cases explictly?

This makes it easier to compare the implementation to the spec.

Reply:
Would not be in favor for the same reasons. Also seems like this approach was reasonably grok-able by someone not familiar with our tests: https://mongodb.slack.com/archives/C0668RJBA0J/p1740413020860229?thread_ts=1740405607.463789&cid=C0668RJBA0J

Reply:
So, I'm not feeling particularly strong enough to continue arguing this, but as someone intimately familiar with our tests, _I_ am telling you that this was annoying to grok.  Especially compared to something like https://github.com/mongodb/node-mongodb-native/blob/cb6a7ba61c112738e08c90330da491a3ab8fdf36/test/manual/search-index-management.prose.test.ts#L138